pkg/alertmanager/amcfg.go

```diff
@@ -1724,6 +1724,11 @@ func (hc *httpClientConfig) sanitize(amVersion semver.Version, logger log.Logger
 		return err
 	}
 
+	if hc.ProxyURL != "" {
+		_, err := url.Parse(hc.ProxyURL)
```

Comment:
taking the idea of validation on proxyurl obj of `oauth2` type would be better. 
https://github.com/prometheus-operator/prometheus-operator/blob/4664990a5e2f15bd1ae408d54bd77cfd90c0c7c6/pkg/alertmanager/amcfg.go#L1783-L1785

cc @simonpasquier

Reply:
Sorry @afzal442 don't follow what you mean 😄 
The code snippet you shared is only about log message.
Could you clarify?

Reply:
Aha, I see. I didn't look into that much. Nice caught. Parsing was missing. I think only one extra condition to be met here is `!amVersion.GTE(semver.MustParse("0.25.0"))`. 🤔